content/php-api-using-jwt/index.md

```diff
@@ -0,0 +1,226 @@
+
+The server and user can verify this signature to know about the data's security and integrity. If this signature matches at both ends, then the data is considered secure, and all other transactions can take place.
+
+As you've known everything about JWT, let's secure your PHP API using JWT. Follow the code along, and you'll have a tamper-proof PHP API.
+
+In this blog, we'll make a simple login page and authenticate it using JWT. This will help you get started with JWT and PHP.
+
+To follow along, you'll need to have PHP and Composer installed on your computer.
+
+If you haven't already installed composer on your computer, you may do so here. Once the composer has been installed, run the tool from your project folder. Composer will assist you in installing Firebase PHP-JWT, a third-party library for working with JWTs and Apache.
+
+Once the library is installed, you'll need to set up a login code in authenticate.php. While you do this, put a code piece that checks and gets the autoloader from the composer tool. The below code will help you achieve this.
+```php
+<?php
+declare(strict_types=1);
+use Firebase\JWT\JWT;
+require_once('../vendor/autoload.php');
+```
+When the form gets submitted, you need to check the entered data with a data source or database. For our purpose, we'll create a hasValidCredentials variable and set it to true. Setting this variable to true means that our data is checked and valid.
+
+```php
+<?php
+// extract credentials from the request
+if ($hasValidCredentials) {
+```
+Any further coding will be wrapped in this block only. The value of the hasValidCredentials variable governs all code related to the production and validation of the required JWT. If its value is true, the JWT shall be created; otherwise, an error will be shown.
+
+Let's start creating our JWT. To begin, we'll need to generate some more variables to aid us in this process. As we saw in the payload section, we must create a variable that will hold our secret key, which may be retrieved from the environment files, another variable to hold information about when the JWT was created, a variable that will hold the JWT's expiration date and time, a username field to identify the client making the authorization request, and a server name variable to register the server name.
+
+JWT's can be easily inspected and checked at client-side browsers. So it is better to hide your secret key and other important information in some environment file, which the user cannot access through client-side requests.
+
+```php
+$secretKey  = 'bGS6lzFqvvSQ8ALbOxatm7/Vk7mLQyzqaS34Q4oR1ew=';
+$issuedAt   = new DateTimeImmutable();
+$expire     = $issuedAt->modify('+6 minutes')->getTimestamp();      // Add 60 seconds
+$serverName = "your.domain.name";
+$username   = "username";                                           // Retrieved from filtered POST data
+$data = [
```

Comment:
This code block is exactly matching the source here: PHP Authorization with JWT (JSON Web Tokens) - SitePoint. https://www.sitepoint.com/php-authorization-jwt-json-web-tokens/

Can you please try to make it unique by modifying the comments and secret key, for example?

Reply:
Done